python/pyspark/sql/pandas/serializers.py

```diff
@@ -216,7 +216,7 @@ def _create_batch(self, series):
             series = [series]
         series = ((s, None) if not isinstance(s, (list, tuple)) else s for s in series)
 
-        def create_array(s, t):
+        def create_array(s, t, n):
```

Comment:
I think you can get the name from `Series` e.g., `s.name` instead of passing extra `n`?

Reply:
I know, this parameter is ugly as it is only used for the exception, but `s.name` is `1` instead of `mean` (test `test_apply_in_pandas_returning_incompatible_type`) when I apply this change.

Looks like `s[field.name]` and `s[s.columns[i]]` returns a `Series` that thinks it's name is `1`. Sad.

Reply:
But `s[s.columns[i]].rename(field.name)` fixes that. Thanks for the idea!

Reply:
Done.